src/client.ts

```diff
@@ -183,4 +194,146 @@ export class Client {
   public getUrl(path: string = ''): string {
     return this.transport.getUrl(path);
   }
+
+  /**
+   * Downloads a file from the store
+   *
+   * The file is placed in the download directory. If the file does not exist,
+   * an error is thrown.
+   *
+   * @param artifact A file object to download
+   * @returns Absolute path to the local copy of the file
+   * @async
+   */
+  public async downloadArtifact(
+    artifact: Artifact,
+    downloadDirectory?: string
+  ): Promise<string> {
+    const downloadDir = downloadDirectory || this.downloadDirectory;
+    if (!downloadDir) {
+      throw new Error('Download directory not specified');
+    }
+
+    const url = this.getUrl(artifact.download_url);
```

Comment:
_nit_: We could add a check if the directory exists and throw if not. However, applications should usually use something like `mkdirp` to create the path on the fly.

Reply:
Added some additional checks that the path exists and that it's a directory.